### PR TITLE
Add SSL flags in extension.ServeREST

### PR DESCRIPTION
**Pull request by jaimeteb**
#12

### CODE REVIEW

**Comment by jaimeteb** [comment]
@ryancurrah what do you think?

I followed these steps for local testing:

```
openssl req -x509 -out localhost.crt -keyout localhost.key \
  -newkey rsa:2048 -nodes -sha256 \
  -subj '/CN=localhost' -extensions EXT -config <( \
   printf "[dn]\nCN=localhost\n[req]\ndistinguished_name = dn\n[EXT]\nsubjectAltName=DNS:localhost\nkeyUsage=digitalSignature\nextendedKeyUsage=serverAuth")
```

```
sudo cp localhost.crt /usr/local/share/ca-certificates/
```

```
sudo dpkg-reconfigure ca-certificates
```

```
go run examples/04_trivia/ext/ext.go -ssl-keyfile localhost.key -ssl-certificate localhost.crt
```

and change REST URL to `https` in ``bot.yml`

File: extension/server.go [hunk 1]
```diff
@@ -113,6 +113,10 @@ type ListenerREST struct {
 func ServeREST(registeredFuncs RegisteredFuncs) error {
 	port := flag.Int("port", 8770, "Port to run extension server on")
 	debug := flag.Bool("debug", false, "Enable debug logging.")
+
+	sslKey := flag.String("ssl-keyfile", "", "SSL key file for TLS secured server.")
+	sslCert := flag.String("ssl-certificate", "", "SSL certificate for TLS secured server.")
```
Comment (ryancurrah) [inline extension/server.go:118]: ssl-cert should suffice here

File: extension/server.go [hunk 2]
```diff
@@ -123,8 +127,13 @@ func ServeREST(registeredFuncs RegisteredFuncs) error {
 	r.HandleFunc("/ext/get_func", l.GetFunc).Methods("POST")
 	r.HandleFunc("/ext/get_all_funcs", l.GetAllFuncs).Methods("GET")
 
-	log.Infof("REST extension server started. Using port %v", *port)
-	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%v", *port), r))
+	if *sslKey != "" && *sslCert != "" {
+		log.Infof("REST extension server started with TLS. Using port %v", *port)
+		log.Fatal(http.ListenAndServeTLS(fmt.Sprintf(":%v", *port), *sslCert, *sslKey, r))
```
Comment (ryancurrah) [inline extension/server.go:132]: Should use %d instead of %v here and below

File: extension/server.go [hunk 3]
```diff
@@ -113,6 +113,10 @@ type ListenerREST struct {
 func ServeREST(registeredFuncs RegisteredFuncs) error {
 	port := flag.Int("port", 8770, "Port to run extension server on")
 	debug := flag.Bool("debug", false, "Enable debug logging.")
+
+	sslKey := flag.String("ssl-keyfile", "", "SSL key file for TLS secured server.")
```
Comment (ryancurrah) [inline extension/server.go:117]: ssl-key should suffice as well

**Review by ryancurrah — APPROVED** [review]
Overall looks good. Made some minor comments

**Comment by jaimeteb** [comment]
> Overall looks good. Made some minor comments

Nice. I'll make those changes tomorrow morning. Thanks!

**Pull request merged**